packages/front-end/components/Features/ConditionInput.tsx

```diff
@@ -173,6 +173,18 @@ export default function ConditionInput(props: Props) {
               return;
             }
 
+            const attributeOptions = filteredAttributeSchema.map((s) => ({
+              label: s.property,
+              value: s.property,
+            }));
+
+            // if the field is not in the attributeOptions list, switch to advanced mode
+            if (
+              !attributeOptions.find((attribute) => attribute.value === field)
+            ) {
+              setAdvanced(true);
```

Comment:
Should not set state during rendering.  Would be better to put this logic in the initial useState value for `advanced`

```ts
  const [advanced, setAdvanced] = useState(
    () => jsonToConds(props.defaultValue, attributes) === null
  );
```

Easiest solution would be to filter `attributes` before passing this in (there's already logic in `jsonToConds` to return `null` when an attribute is missing there)

Reply:
@jdorn Ah - I see - that makes sense.

I just pushed an update in this commit that moves the filtering logic to the useAttributeSchema hook. Let me know what you think of that pattern.

After looking in the app, it looks like there are a few other places we might want to filter the attributes based on the project (experiment assignment selector, feature flag testing tool, etc). That can be done in a separate PR, I just wanted to call it out.